README.md

```diff
@@ -17,6 +17,7 @@ Table of contents
 * [Eloquent](#eloquent)
 * [Optional: Alias](#optional-alias)
 * [Query Builder](#query-builder)
+* [`toSql()`](#to-sql)
```

Comment:
Maybe we can put this under a category called "debugging" or "inspecting queries".

Reply:
Add else info about DB::getQueryLog()

Reply:
It old feature but getQueryLog() return original query struncture to mongodb

Reply:
@jenssegers maybe fix this after #1917

Reply:
@Smolevich i will look into it as soon as i get home

Reply:
Found a solution but some of the code from `getFresh()` needs refactoring.

Reply:
I suggest leaving this PR as-is, in the SQL form, because the `getFresh()` needs a lot of refactoring.

Reply:
What `getFresh()` is used for? @rennokki

Reply:
It is used to compile the wheres, options and get data for the `get()` method.

There are some edge cases, like distinct or aggregations, both checks are in `getFresh()`. I assume that the best way is to compile the query in the builder and pass it to the `Collection` class where the call is actually made to the `MongoCollection`'s methods that execute the needed query. 🤷‍♂

Reply:
@rennokki what about to add info about using `DB::getQueryLog()`
For example 

```
DB::enableQueryLog();
$user = factory(User::class)->create();
$query = DB::connection()->table(config('auth.passwords.users.table'));
dd(DB::getQueryLog());
```
Example of element of array that function `getQueryLog` returned 

```
1 => array:3 [
    "query" => "users.find({"email":"oziemann@example.org"},{"limit":1,"typeMap":{"root":"array","document":"array"}})"
    "bindings" => []
    "time" => 4.49
  ]
```

Reply:
The scope of this PR was to be able to get the query string before the actual query was executed, so i can be able to add cache query-by-query.

I can add the query log thing in the readme, because that's useful.

Reply:
So far, the single issue here is that the basic `get()` and the aggregate and groups are treated differently. This means that the final query is split into two methods rather than a single `get()` one.

I repeat - the purpose of this PR was to get the string for each query, so I can be able to query the cache query-by-query.

Reply:
@Smolevich @Giacomo92 

So far, the single issue here is that the basic `get()` and the aggregate and groups are treated differently. This means that the final query is split into two methods rather than a single `get()` one.

I repeat - the purpose of this PR was to get the string for each query, so I can be able to query the cache query-by-query.

Reply:
@rennokki, do you see this method https://github.com/jenssegers/laravel-mongodb/blob/a8918acadb79f4f90d42eaddf0e996054ddd8184/src/Jenssegers/Mongodb/Query/Builder.php#L413?

Reply:
@Smolevich I think `$options` and `$pipeline` has to be attributes in the class and be passed to the `generateCacheKey` method.